lib/model-builder.js

```diff
@@ -308,7 +308,9 @@ ModelBuilder.prototype.define = function defineClass(className, properties, sett
   // Add the id property
   if (idInjection) {
     // Set up the id property
-    ModelClass.definition.defineProperty('id', {type: Number, id: 1, generated: true});
+    // updateOnly property is added to indicate that this property will appear in the model for update/updateorcreate
+    // operations but and not for create operation.
+    ModelClass.definition.defineProperty('id', {type: Number, id: 1, generated: true, updateOnly: true});
```

Comment:
I think `updateOnly` should be set only when `forceId` is in effect. With `forceId:false`, clients can provide custom `id` value even when the `id` column was injected by the datasource.

I think `updateOnly` flag should be set in `lib/datasource.js`:

```js
if (forceId) {
  modelClass.validatesAbsenceOf(idName, {if: 'isNewRecord'});
  // set updateOnly here
  // something along the following lines, I am not sure if we need to change rawProperties too
  modelClass.definition.properties[idName].updateOnly = true;
}
```